### ISSUE TITLE
Enhance NaturalKeyOrPKMultipleChoiceFilter to handle both UUID and Integer pk field

### ISSUE THREAD

**Issue by timizuoebideri1** [nautobot/nautobot#3336]

**Comment by bryanculver** [comment]
@nautobot/core I belive the only cases for this are Django core models (and users, groups). Are there more side-effects that would have to be addressed with this than consirding it a valid limitation for those models? Maybe eventually renaming to `NaturalKeyOrUUIDPKMultipleChoiceFilter`?

**Comment by glennmatthews** [comment]
Yeah, this is specifically for filtering on models like `Group` and `ContentType` that aren't Nautobot models and so have integer PKs. We have to keep the old separation of `group` (natural-key) versus `group_id` (integer PK) filters for those models at present.

It's trickier than it sounds to fix this though, because a string like "123" could be a PK **or** a slug (or other natural key, like `VLAN.vid`), unlike the UUID case where if something looks like a UUID, the odds that it's "really" a natural key are low enough that we haven't bothered to care about that case - so the filtering logic would have to become a decent amount smarter in order to handle this case intelligently.

**Comment by bryanculver** [comment]
> Yeah, this is specifically for filtering on models like `Group` and `ContentType` that aren't Nautobot models and so have integer PKs. We have to keep the old separation of `group` (natural-key) versus `group_id` (integer PK) filters for those models at present.
> 
> It's trickier than it sounds to fix this though, because a string like "123" could be a PK **or** a slug (or other natural key, like `VLAN.vid`), unlike the UUID case where if something looks like a UUID, the odds that it's "really" a natural key are low enough that we haven't bothered to care about that case - so the filtering logic would have to become a decent amount smarter in order to handle this case intelligently.

So is it worth making one filter for every possible combination or in those cases using the more generic/specific multiple choice (no natural key support, or knowing that the target models likely won’t have integers for natural keys)?
